fix: adjust OrderCreation.spec.js in order not to update write-protected line-item properties, increase minimum version for feature toggle route and adjust integration test routine

philipreinken opened the pull request:
chore(ci): update compose.yaml
chore(ci): add shopware.Dockerfile for test image builds
chore(ci): adjust test workflow


Refs: https://github.com/shopware/acceptance-test-suite/issues/313

philipreinken committed:
chore(ci): adjust integration test routine

chore(ci): update compose.yaml
chore(ci): add shopware.Dockerfile for test image builds
chore(ci): adjust test workflow

Refs: https://github.com/shopware/acceptance-test-suite/issues/313

philipreinken committed:
chore(ci): add a job name

philipreinken committed:
chore(ci): rename to test-image

pweyck commented on compose.yaml:
```diff
@@ -45,50 +38,78 @@ x-shopware: &shopware-65
+  init:
+    <<: [ *service, *shopware-volumes ]
+    image: ghcr.io/shopware/acceptance-test-suite/test-image:${SHOPWARE_VERSION:-trunk}
+    build:
+      context: .
+      dockerfile: test-image.Dockerfile
+      args:
+        - PHP_VERSION=${PHP_VERSION:-8.4}
+        - SHOPWARE_VERSION=${SHOPWARE_VERSION:-trunk}
+        - SHOPWARE_BUILD_SOURCE=${SHOPWARE_BUILD_SOURCE:-branch}
```
```suggestion
```
The images are build in the test jobs right. We should only build them in the nightly

philipreinken replied:
This is useful for regular usage of the `compose.yaml`, so I'd like to keep it; once the images have been registered/pushed by the nightly once, they will not be built anymore, but only pulled by the test job.

pweyck commented on test-image.Dockerfile:
```diff
@@ -0,0 +1,74 @@
+
+ARG SHOPWARE_VERSION
+
+ENV COMPOSER_ROOT_VERSION="6.7.9999999-dev"
```
this should not be necessary, except for trunk. For other versions we should use the tags

philipreinken replied:
This only comes into play when checking out a development branch (for example trunk). [Tags are set up](https://github.com/shopware/acceptance-test-suite/pull/334/files/4a1a21e26f482dcab172a46786c00a8017cd2def#diff-01f0199904147627781673f29653c3c3430acb09b5cbbe5d2f5d454e13c6e6eeR11) using `shopware-cli project create [version]`.

But I just realised that `6.6.x` will also use this value in the current state, so I could make this configurable as well. Sounds alright?

pweyck commented on .github/workflows/build-shopware-image.yml:
```diff
@@ -0,0 +1,49 @@
+  workflow_dispatch:
+
+jobs:
+  build-test-image:
```
we should build the image for different released versions addition to 6.6.x and trunk branch. You can use the action [shopware/github-actions/versions@main](https://github.com/shopware/shopware/blob/e3c4e85f881e18141bfb87a9ff261afb33e6e5a0/.github/workflows/integration.yml#L330-L352)

The ats should only test against released versions not trunk (because it should continue to work with older versions, otherwise it's pretty useless for externals)

philipreinken replied:
Makes sense; I'll add those test runs 👍

philipreinken committed:
chore(ci): build more shopware versions

chore(ci): test released versions only

philipreinken committed:
chore(ci): fix v6.5.0.0 build

philipreinken committed:
chore(ci): provide pull access to ghcr.io

philipreinken committed:
chore(ci): allow configuration of COMPOSER_ROOT_VERSION for branch-based builds

philipreinken committed:
chore(ci): adjust version constraint for feature availability check

The corresponding route was not part of the v6.6.0.0 release, but introduced later in v6.6.1.0

Refs: https://github.com/shopware/shopware/commit/85eddda8a18096d2a98005b4c83c4965e57dadc4#diff-070bbddb3348a48c0b0504b62495d82049b54293df5d52d181f3f7834a556901R46

philipreinken commented on src/fixtures/HelperFixtures.ts:
```diff
@@ -68,7 +68,7 @@ export const test = base.extend<NonNullable<unknown>, FixtureTypes>({
 
             const features: FeaturesType = {};
-            if (satisfies(config.version, '>=6.6.0.0')) {
+            if (satisfies(config.version, '>=6.6.1.0')) {
```
@pweyck it looks like the `/_action/feature-flag` route was introduced with `v6.6.1.0` and is not present in `v6.6.0.0`.

philipreinken replied:
Not sure if I break something by increasing this, but it will fail otherwise when executed with `v6.6.0.0`.

philipreinken committed:
chore(debug): don't send promotionId as part of the promotion line item

philipreinken committed:
chore(test): adjust DataSharing page object to take differences in versions <6.6.1.0 into account

philipreinken committed:
chore(ci): rename test job

philipreinken commented:
Adjustments to the required checks: https://github.com/shopware/organization/pull/295

philipreinken committed:
chore(test): unset write-protected promotionId in lineItem struct prior to sending it off

pweyck commented on compose.yaml:
```diff
@@ -1,42 +1,35 @@
 x-env: &env
```
I think we should also update the documentation on how to run it locally

philipreinken replied:
I've adjusted the docs 😊

philipreinken committed:
docs: add documentation regarding the changed local setup

yusufttur commented on .github/workflows/build-shopware-image.yml:
```diff
@@ -0,0 +1,74 @@
+    name: Determine Shopware versions to build
+    runs-on: ubuntu-24.04
+    outputs:
+      first-version: ${{ steps.versions.outputs.first-version }}
+      latest-version: ${{ steps.versions.outputs.latest-version }}
+      lts-first-version: ${{ steps.versions.outputs.lts-first-version }}
+      lts-latest-version: ${{ steps.versions.outputs.lts-latest-version }}
```
Can you explain a bit what these versions mean?

philipreinken replied:
Ah yes, I've just copied those over, I should have added some comments:
- `first`: First version of the current major, e.g. `v6.6.0.0`
- `latest`: Latest released version of the current major, e.g. `v6.6.10.1`
- `lts-first`: First version of the previous major, e.g. `v6.5.0.0`
- `lts-latest`: Latest version of the previous major, e.g. `v6.5.8.16`

philipreinken committed:
docs(ci): add comments regarding version identifiers
